src/linear_algebra/general_linear_group.lean

```diff
@@ -186,11 +189,26 @@ lemma to_GL_pos_injective :
 (show function.injective ((coe : GL_pos n R → matrix n n R) ∘ to_GL_pos),
  from subtype.coe_injective).of_comp
 
+/-- Coercing a `special_linear_group` via `GL_pos` and `GL` is the same as coercing striaght to a
+matrix -/
+@[simp]
+lemma coe_GL_pos_coe_GL_coe_matrix (g : special_linear_group n R) :
+  (↑(↑(↑(g : special_linear_group n R) : GL_pos n R) : GL n R) : matrix n n R) = ↑g := rfl
+
+@[simp] lemma coe_to_GL_pos_to_GL_det (g : special_linear_group n R) :
+  ((g : GL_pos n R) : GL n R).det = 1 :=
+units.ext g.prop
+
+variable [fact (even (fintype.card n))]
+
+@[norm_cast] lemma coe_GL_pos_neg (g : special_linear_group n R) :
+  ↑(-g) = -(↑g : GL_pos n R) := subtype.ext $ units.ext rfl
```

Comment:
```suggestion
  ↑(-g) = -(g : GL_pos n R) := subtype.ext $ units.ext rfl
```
I am not totally sure, but if you specify the type there is no need of writing `↑` (similarly elsewhere).

Reply:
Sometimes I've had reviewers say they prefer the explicit `↑` as it makes it obvious that a coercion is happening.

Reply:
I am not against it, but we may want to fix a convention... in any case it is not so important.

Reply:
I sort of like having the `↑` but I'm happy to change it if needed.